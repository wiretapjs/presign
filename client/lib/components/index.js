export { default as Button } from './Button/Button'
export { default as Background } from './Background/Background'
export { default as Card } from './Card/Card'
export { default as ErrorMessage } from './ErrorMessage/ErrorMessage'
export { default as Paper } from './Paper/Paper'
export { default as EnvironmentWatermark } from './EnvironmentWatermark/EnvironmentWatermark'
export { default as Header } from './Header/Header'
export { default as LegalText } from './LegalText/LegalText'
export { default as Logo } from './Logo/Logo'
export { default as MainContainer } from './MainContainer/MainContainer'
export { default as MainTitle } from './MainTitle/MainTitle'
export { default as Column } from './Column/Column'
export { default as Row } from './Row/Row'
export { default as Span } from './Span/Span'
export { default as Footer } from './Footer/Footer'
export { default as EsignIframe } from './Iframe/EsignIframe'
export { default as HelpMessage } from './HelpMessage/HelpMessage'
