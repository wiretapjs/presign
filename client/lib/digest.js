export default function (string) {
  return string.split('').reduce((accumulator, character) => {
    accumulator = ((accumulator << 5) - accumulator) + character.charCodeAt(0)
    return accumulator & accumulator
  }, 0)
}
