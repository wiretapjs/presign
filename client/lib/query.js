import QueryString from 'query-string'

const get = function (param) {
  return QueryString.parse(window.location.search)[param]
}

export default { get }
