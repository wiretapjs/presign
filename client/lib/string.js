export const capitalize = function (string) {
  if (!string) throw new Error('Undefined input')
  return string.charAt(0).toUpperCase() + string.slice(1)
}
