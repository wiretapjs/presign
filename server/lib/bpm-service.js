const request = require('request')
const HttpStatus = require('http-status-codes')
const _ = require('lodash')
const Url = require('url-parse')
const RemoteServiceError = require('models/errors/remoteService')
const config = require('../config')

const DEFAULT_REQUEST_OPTIONS = {
  json: true,
  proxy: config.proxy,
}

const isErrorResponse = function (err, response) {
  return err || response.statusCode >= HttpStatus.BAD_REQUEST || _.get(response, 'body.error')
}

const formatRequest = function (options) {
  const requestConfig = Object.assign({}, DEFAULT_REQUEST_OPTIONS, options)
  requestConfig.headers.host = getRequestHost(requestConfig.url)

  return requestConfig
}

const invoke = function (requestConfig) {
  requestConfig = formatRequest(requestConfig)
  return new Promise((resolve, reject) => {
    request(requestConfig, (err, response) => {
      if (isErrorResponse(err, response)) {
        return reject(new RemoteServiceError(`Failed to get ${requestConfig.url}`, {
          response,
          internalError: err,
          headers: _.get(response, 'headers'),
        }))
      }
      return resolve(response)
    })
  })
}

const get = function (requestConfig) {
  requestConfig.method = 'GET'
  return invoke(requestConfig)
}

const getRequestHost = function (requestUrl) {
  return new Url(requestUrl).host
}

module.exports = { get, invoke }
