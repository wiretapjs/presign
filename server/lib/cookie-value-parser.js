class CookieValueParser {
  filterCookies (cookies, filter) {
    if (!cookies || !filter) {
      return []
    }

    const filteredCookies = cookies.filter(cookie => cookie.toLowerCase()
      .includes(filter.toLowerCase()))

    return filteredCookies
  }
}

module.exports = new CookieValueParser()
