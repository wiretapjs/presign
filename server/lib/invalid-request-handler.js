const ResultEnvelope = require('models/resultEnvelope')
const InvalidRequestError = require('models/errors/invalidRequest')

const handleInvalidRequest = function (validationResponse) {
  return Promise.reject(ResultEnvelope.Error(
    new InvalidRequestError(
      `Validation Errors occured when validating your request.`,
      {
        validationErrors: validationResponse.errors,
      })))
}

module.exports = handleInvalidRequest
