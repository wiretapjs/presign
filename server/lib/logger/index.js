const pino = require('pino')()

const sendMessage = message => {
  const { type = 'info', text = '' } = message
  if (text) {
    if (type === 'error') pino.error(text)
    pino.info(text)
  }
  return message
}

module.exports = {
  sendMessage,
}
