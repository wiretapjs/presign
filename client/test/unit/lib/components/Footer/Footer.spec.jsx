import React from 'react'
import { shallow } from 'enzyme'
import { Footer } from 'lib/components'

describe('Footer', () => {
  it('should render correctly', () => {
    const ahmsg = shallow(<Footer />)
    expect(ahmsg).toMatchSnapshot()
  })
})
