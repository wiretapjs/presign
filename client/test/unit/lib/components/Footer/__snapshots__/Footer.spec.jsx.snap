// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.div>
      <styled.a
        href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138362980102"
      >
        Privacy Policy
      </styled.a>
       | 
      <styled.a
        href="http://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1240002784964"
      >
        Security Policy
      </styled.a>
       | 
      <styled.a
        href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138366154501"
      >
        Terms & Conditions
      </styled.a>
       | 
      <styled.a
        href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138366406840"
      >
        Fraud Protection
      </styled.a>
       | 
      <styled.a
        href="http://public.libertymutual-cdn.com/PmEService/MediaAssets/pages/terms-and-conditions-enrollment.html"
      >
        eService & Paperless Terms & Conditions
      </styled.a>
    </styled.div>
    <styled.div>
      © 2017 Liberty Mutual Insurance Company, 175 Berkeley Street, Boston, MA 02116
    </styled.div>
  </styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div>
        <styled.a
          href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138362980102"
        >
          Privacy Policy
        </styled.a>
         | 
        <styled.a
          href="http://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1240002784964"
        >
          Security Policy
        </styled.a>
         | 
        <styled.a
          href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138366154501"
        >
          Terms & Conditions
        </styled.a>
         | 
        <styled.a
          href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138366406840"
        >
          Fraud Protection
        </styled.a>
         | 
        <styled.a
          href="http://public.libertymutual-cdn.com/PmEService/MediaAssets/pages/terms-and-conditions-enrollment.html"
        >
          eService & Paperless Terms & Conditions
        </styled.a>
      </styled.div>
      <styled.div>
        © 2017 Liberty Mutual Insurance Company, 175 Berkeley Street, Boston, MA 02116
      </styled.div>
    </styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Footer />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Footer {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.div>
            <styled.a
              href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138362980102"
            >
              Privacy Policy
            </styled.a>
             | 
            <styled.a
              href="http://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1240002784964"
            >
              Security Policy
            </styled.a>
             | 
            <styled.a
              href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138366154501"
            >
              Terms & Conditions
            </styled.a>
             | 
            <styled.a
              href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138366406840"
            >
              Fraud Protection
            </styled.a>
             | 
            <styled.a
              href="http://public.libertymutual-cdn.com/PmEService/MediaAssets/pages/terms-and-conditions-enrollment.html"
            >
              eService & Paperless Terms & Conditions
            </styled.a>
          </styled.div>
          <styled.div>
            © 2017 Liberty Mutual Insurance Company, 175 Berkeley Street, Boston, MA 02116
          </styled.div>
        </styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.div>
            <styled.a
              href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138362980102"
            >
              Privacy Policy
            </styled.a>
             | 
            <styled.a
              href="http://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1240002784964"
            >
              Security Policy
            </styled.a>
             | 
            <styled.a
              href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138366154501"
            >
              Terms & Conditions
            </styled.a>
             | 
            <styled.a
              href="https://www.libertymutualgroup.com/omapps/ContentServer?pagename=LMGroup/Views/lmgDisclosure&cid=1138366406840"
            >
              Fraud Protection
            </styled.a>
             | 
            <styled.a
              href="http://public.libertymutual-cdn.com/PmEService/MediaAssets/pages/terms-and-conditions-enrollment.html"
            >
              eService & Paperless Terms & Conditions
            </styled.a>
          </styled.div>
          <styled.div>
            © 2017 Liberty Mutual Insurance Company, 175 Berkeley Street, Boston, MA 02116
          </styled.div>
        </styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Footer />,
}
`;
